### PR TITLE
3/palette prep

### PR DESCRIPTION
**Pull request by stuartromanek**
This work supports the integration of `@apostrophecms/palette`

- adds admin bar dimensions to window object for easy cross module sizing
- add various icons
- add `displayOptions` object to schemas which can inform schema ui changes
- `color` input: work around no null values for `vue-color`
- add `micro` and `inline` style modifiers for schema displays

### CODE REVIEW

File: modules/@apostrophecms/schema/index.js [hunk 1]
```diff
@@ -91,7 +91,7 @@ module.exports = {
         // This is new and until now if JS client side failed, then it would
         // allow the save with empty values -Lars
         if (field.required && (_.isUndefined(data[field.name]) || !data[field.name].toString().length)) {
-          throw self.apos.error('required');
+          throw self.apos.error(`required: ${field.name}`);
```
Comment (abea) [inline modules/@apostrophecms/schema/index.js:94]: `'required'` is a specific response that the tests look for. You can add a more helpful message as a second argument, like here: https://github.com/apostrophecms/apostrophe/blob/3/palette-prep/modules/@apostrophecms/attachment/index.js#L433

File: modules/@apostrophecms/ui/ui/apos/scss/global/_inputs.scss [hunk 2]
```diff
@@ -95,6 +95,56 @@
...
+    }
+    .apos-color__info {
+      font-size: var(--a-type-small);
+      margin-left: 5px;
+      white-space: nowrap;
+    }
+  }
+}
+
+// .apos-field-help-tooltip {
```
Comment (abea) [inline modules/@apostrophecms/ui/ui/apos/scss/global/_inputs.scss]: stray bits

File: modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue [hunk 3]
```diff
@@ -1,22 +1,37 @@
...
+          class="apos-field-label"
+          :is="labelEl" :for="uid"
+        >
+          {{ field.label }}
+          <span v-if="field.required" class="apos-field-required">
+            *
+          </span>
+          <span
+            v-if="field.help && displayOptions.helpTooltip"
+            class="apos-field-help-tooltip"
```
Comment (abea) [inline modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue:17]: This component seems to be totally abandoning BEM.

File: modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue [hunk 4]
```diff
@@ -83,6 +104,9 @@ export default {
           classes.push(`apos-field--${m}`);
         });
       }
+      if (this.displayOptions && this.displayOptions.helpTooltip) {
+        classes.push('apos-field--help-tooltip');
```
Comment (abea) [inline modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue:108]: Is this class used?

File: modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue [hunk 5]
```diff
@@ -124,43 +148,5 @@ export default {
 </script>
 
 <style lang="scss" scoped>
-  .apos-field {
```
Comment (abea) [inline modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue:127]: Why move these. Are they used by another? All inputs are wrapped by this.

**Review by abea — CHANGES_REQUESTED** [review]
dang it. Those were supposed to be change requests.

File: modules/@apostrophecms/schema/index.js (hunk 1 above)
Comment (stuartromanek) [reply modules/@apostrophecms/schema/index.js:94]: oo didn't mean to commit this, for my own debugging

File: modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue (hunk 4 above)
Comment (stuartromanek) [reply modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue:108]: n

File: modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue (hunk 3 above)
Comment (stuartromanek) [reply modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue:17]: re-bem

File: modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue (hunk 5 above)
Comment (stuartromanek) [reply modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue:131]: moved back

File: modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue [hunk 6]
```diff
@@ -1,29 +1,41 @@
...
+              :icon-size="11"
+              icon-color="var(--a-base-4)"
+            />
+          </span>
+        </component>
+        <!-- TODO i18n -->
+        <p
+          v-if="(field.help || field.htmlHelp) && !displayOptions.helpTooltip"
+          class="apos-field__help"
+          v-html="(field.help || field.htmlHelp)"
```
Comment (stuartromanek) [inline modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue:32]: @abea do note the consolidation of this help UI, its new in the most recent merge (tested and working to put HTML into the tooltip bubble as well)

**Review by abea — APPROVED** [review]

**Pull request merged**